selftests/check.py

```diff
@@ -518,6 +519,19 @@ def create_suites(args):
     return suites
 
 
+def print_failed_tests(job_log_dir):
+    jfile = os.path.join(job_log_dir, '.journal.sqlite')
+    db = sqlite3.connect(jfile)
+    cur = db.cursor()
+    cur.execute("SELECT * FROM test_journal WHERE status='FAIL' OR"
+                " status='ERROR';")
```

Comment:
Is journal a "required plugin"? I was thinking more about getting it directly from the job/tests list. And make this more abstract.  Depending on SQL syntax there might be a yellow flag for future decoupling.
 
In my mind we could do something like this:

```
job = Job()
job.run()
failed = job.get_failed_tests()
```

But of course that I didn't have the chance to see the problems with this approach, so take it with a grant of salt.

Reply:
I get your point, true is that I didn't consider the future decoupling, but I don't want to create a new feature because of this. I want to use existing features to show how it can be used. Because I think that if we have a usage for some plugin in our testing we should use it.

Reply:
And when we will do the decoupling the journal plugin can be as `selftest-requirements`

Reply:
Having an usage for a plugin IMO should not be the motivation. But I understand some of the motivation, so I will not block this, of course. But have you tried to use ` job.result.tests`? Any reason for not using this?

Like I said, if possible I would like to use our objects and proprieties instead of hardcoded SQL. This gives us more flexibility.

Reply:
I'm not against using plugins to run our self-tests. Like @clebergnu mentioned once to me, this is a nice place to show Avocado's capabilities.

OTOH, I agree that we should avoid adding dependencies to the Avocado core other than pure python packages.

Reply:
> Having an usage for a plugin IMO should not be the motivation. But I understand some of the motivation, so I will not block this, of course. But have you tried to use ` job.result.tests`? Any reason for not using this?

@beraldoleal as I said my motivation was to create and example of usage of existing plugin, but of course I can use `job.result.tests` and create `job.get_failed_tests()`. I just think that we should show what avocado plugins can do.

Reply:
@richtja I agree that we should show off our plugins and our potential. But IMO this is a bad example. Redirecting users to use SQL instead of nice python attributes, could be discouraging for new users.

Reply:
Ok, I think you are right, it's too much complicated. I will create a new version.

Reply:
> Ok, I think you are right, it's too much complicated. I will create a new version.

If you don't mind, I think we can live with this version until we have time to create, and properly document, a results API that is sound enough and well documented.  I don't think it will me a trivial task.